### ISSUE TITLE
add scala.js/native support

### ISSUE THREAD

**Issue by antonkulaga** [kailuowang/henkan#45]
The library will be useful not only in jvm but also in .js/.native/.shared projects

**Comment by kailuowang** [comment]
agree, shapeless is adding native support, so hopefully it should be ready soon. As for scala.js we already have an issue https://github.com/kailuowang/henkan/issues/4. Just never get the time to make it happen. Anyone interested?
